src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMTE_FrothFlotationCell.java

```diff
@@ -275,26 +279,38 @@ public boolean checkRecipeGeneric(
 		
 		/*
 		 * 
-		 * Material Hash checks
+		 * Material checks
 		 * Makes sure we can only ever use one type of material in this flotation cell.
+		 * We used to depend on Alk's hash, but it's unreliable and user-hostile
+		 * So we're using unlocalized name of material now.
 		 * 
 		 */
-		int aExpectedMaterialHash;
-		// Set the hash of expected material type
-		if (mLockedOreType == -1) {
-			mLockedOreType = FlotationRecipeHandler.getHashForMaterial(FlotationRecipeHandler.getMaterialOfMilledProduct(FlotationRecipeHandler.findMilledStack(aRecipe)));
+		Material foundMaterial = FlotationRecipeHandler.getMaterialOfMilledProduct(FlotationRecipeHandler.findMilledStack(tRecipe));
+		String foundMaterialName = null;
+		if (foundMaterial != null) {
+			foundMaterialName = foundMaterial.getUnlocalizedName();
 		}
-		// Set the hash for this recipe check
-		aExpectedMaterialHash = mLockedOreType;
-		
-		// Compute hash of current inputs
-		int aFoundMaterialHash = FlotationRecipeHandler.getHashForMaterial(FlotationRecipeHandler.getMaterialOfMilledProduct(FlotationRecipeHandler.findMilledStack(aItemInputs)));
-		
-		// Check hashes match
-		if (aExpectedMaterialHash != aFoundMaterialHash) {
+
+		// Set material locked for this controller
+		if (lockedMaterialName == null) {
+			if (foundMaterialName != null) {
+				lockedMaterialName = foundMaterialName;
+			}
+		}
+
+		// Check material match
+		if (foundMaterialName == null) {
```

Comment:
This can probably be moved above `lockedMaterialName` null check and nested `foundMaterialName` null check then can be omitted.